README.rst
Python Client for Dataproc Metastore API
========================================

|stable| |pypi| |versions|

`Dataproc Metastore API`_: 

- `Client Library Documentation`_
- `Product Documentation`_

.. |stable| image:: https://img.shields.io/badge/support-stable-gold.svg
   :target: https://github.com/googleapis/google-cloud-python/blob/main/README.rst#stability-levels
.. |pypi| image:: https://img.shields.io/pypi/v/google-cloud-dataproc-metastore.svg
   :target: https://pypi.org/project/google-cloud-dataproc-metastore/
.. |versions| image:: https://img.shields.io/pypi/pyversions/google-cloud-dataproc-metastore.svg
   :target: https://pypi.org/project/google-cloud-dataproc-metastore/
.. _Dataproc Metastore API: https://cloud.google.com/dataproc-metastore/
.. _Client Library Documentation: https://cloud.google.com/python/docs/reference/metastore/latest
.. _Product Documentation:  https://cloud.google.com/dataproc-metastore/

Quick Start
-----------

In order to use this library, you first need to go through the following steps:

1. `Select or create a Cloud Platform project.`_
2. `Enable billing for your project.`_
3. `Enable the Dataproc Metastore API.`_
4. `Setup Authentication.`_

.. _Select or create a Cloud Platform project.: https://console.cloud.google.com/project
.. _Enable billing for your project.: https://cloud.google.com/billing/docs/how-to/modify-project#enable_billing_for_a_project
.. _Enable the Dataproc Metastore API.:  https://cloud.google.com/dataproc-metastore/
.. _Setup Authentication.: https://googleapis.dev/python/google-api-core/latest/auth.html

Installation
~~~~~~~~~~~~

Install this library in a `virtualenv`_ using pip. `virtualenv`_ is a tool to
create isolated Python environments. The basic problem it addresses is one of
dependencies and versions, and indirectly permissions.

With `virtualenv`_, it's possible to install this library without needing system
install permissions, and without clashing with the installed system
dependencies.

.. _`virtualenv`: https://virtualenv.pypa.io/en/latest/


Code samples and snippets
~~~~~~~~~~~~~~~~~~~~~~~~~

Code samples and snippets live in the `samples/` folder.


Supported Python Versions
^^^^^^^^^^^^^^^^^^^^^^^^^
Our client libraries are compatible with all current [active](https://devguide.python.org/devcycle/#in-development-main-branch) and [maintenance](https://devguide.python.org/devcycle/#maintenance-branches) versions of
Python.

Python >= 3.7

Unsupported Python Versions
^^^^^^^^^^^^^^^^^^^^^^^^^^^
Python <= 3.6

If you are using an [end-of-life](https://devguide.python.org/devcycle/#end-of-life-branches)
version of Python, we recommend that you update as soon as possible to an actively supported version.


Mac/Linux
^^^^^^^^^

.. code-block:: console

    pip install virtualenv
    virtualenv <your-env>
    source <your-env>/bin/activate
    <your-env>/bin/pip install google-cloud-dataproc-metastore


Windows
^^^^^^^

.. code-block:: console

    pip install virtualenv
    virtualenv <your-env>
    <your-env>\Scripts\activate
    <your-env>\Scripts\pip.exe install google-cloud-dataproc-metastore

Next Steps
~~~~~~~~~~

-  Read the `Client Library Documentation`_ for Dataproc Metastore API
   to see other available methods on the client.
-  Read the `Dataproc Metastore API Product documentation`_ to learn
   more about the product and see How-to Guides.
-  View this `README`_ to see the full list of Cloud
   APIs that we cover.

.. _Dataproc Metastore API Product documentation:  https://cloud.google.com/dataproc-metastore/
.. _README: https://github.com/googleapis/google-cloud-python/blob/main/README.rst


CHANGELOG.md
# Changelog

## [0.3.3](https://github.com/googleapis/python-dataproc-metastore/compare/v0.3.2...v0.3.3) (2022-06-07)


### Bug Fixes

* **deps:** require protobuf<4.0.0 on v0 branch ([#152](https://github.com/googleapis/python-dataproc-metastore/issues/152)) ([b213839](https://github.com/googleapis/python-dataproc-metastore/commit/b21383970961946521357bb5674716beff828e7b))

### [0.3.2](https://github.com/googleapis/python-dataproc-metastore/compare/v0.3.1...v0.3.2) (2022-04-01)


### Bug Fixes

* **deps:** require google-api-core >= 1.31.5, >= 2.3.2 on v0 release ([#127](https://github.com/googleapis/python-dataproc-metastore/issues/127)) ([b1b65e8](https://github.com/googleapis/python-dataproc-metastore/commit/b1b65e84e595768fd4ae324f4fa70b46eeada5b8))

### [0.3.1](https://www.github.com/googleapis/python-dataproc-metastore/compare/v0.3.0...v0.3.1) (2021-07-26)


### Bug Fixes

* **deps:** pin 'google-{api,cloud}-core', 'google-auth' to allow 2.x versions ([#50](https://www.github.com/googleapis/python-dataproc-metastore/issues/50)) ([091ff2f](https://www.github.com/googleapis/python-dataproc-metastore/commit/091ff2fa0cc9413c99cb3c17a18af9de131bca01))
* enable self signed jwt for grpc ([#56](https://www.github.com/googleapis/python-dataproc-metastore/issues/56)) ([3f94f5a](https://www.github.com/googleapis/python-dataproc-metastore/commit/3f94f5adb30d4e9a6e28424259a9a26b78429740))


### Documentation

* add Samples section to CONTRIBUTING.rst ([#51](https://www.github.com/googleapis/python-dataproc-metastore/issues/51)) ([c093c12](https://www.github.com/googleapis/python-dataproc-metastore/commit/c093c1282e832f3d7a027d63be1b55017bcec9ff))


### Miscellaneous Chores

* release 0.3.1 ([#55](https://www.github.com/googleapis/python-dataproc-metastore/issues/55)) ([2a846dd](https://www.github.com/googleapis/python-dataproc-metastore/commit/2a846ddef298a09baf7ff27331cd438f8f7113ee))

## [0.3.0](https://www.github.com/googleapis/python-dataproc-metastore/compare/v0.2.2...v0.3.0) (2021-07-01)


### Features

* add always_use_jwt_access ([#43](https://www.github.com/googleapis/python-dataproc-metastore/issues/43)) ([75cf2ee](https://www.github.com/googleapis/python-dataproc-metastore/commit/75cf2ee2204211be6f43d94bf78cfa7f02ba1976))


### Bug Fixes

* disable always_use_jwt_access ([#47](https://www.github.com/googleapis/python-dataproc-metastore/issues/47)) ([903b08e](https://www.github.com/googleapis/python-dataproc-metastore/commit/903b08e9436691a92f5557d3e8a0a49612d4d8db))


### Documentation

* omit mention of Python 2.7 in 'CONTRIBUTING.rst' ([#1127](https://www.github.com/googleapis/python-dataproc-metastore/issues/1127)) ([#38](https://www.github.com/googleapis/python-dataproc-metastore/issues/38)) ([9b8c147](https://www.github.com/googleapis/python-dataproc-metastore/commit/9b8c14739b9cb5d02f9372d952acf099712f9826)), closes [#1126](https://www.github.com/googleapis/python-dataproc-metastore/issues/1126)

### [0.2.2](https://www.github.com/googleapis/python-dataproc-metastore/compare/v0.2.1...v0.2.2) (2021-06-16)


### Bug Fixes

* **deps:** add packaging requirement ([#35](https://www.github.com/googleapis/python-dataproc-metastore/issues/35)) ([922536c](https://www.github.com/googleapis/python-dataproc-metastore/commit/922536c93fe70eb0052843c6cb9f9a7c91046a81))

### [0.2.1](https://www.github.com/googleapis/python-dataproc-metastore/compare/v0.2.0...v0.2.1) (2021-06-16)


### Bug Fixes

* exclude docs and tests from package ([#33](https://www.github.com/googleapis/python-dataproc-metastore/issues/33)) ([dfaec68](https://www.github.com/googleapis/python-dataproc-metastore/commit/dfaec68833ded607fd0514d73b10e0d33dc26c72))

## [0.2.0](https://www.github.com/googleapis/python-dataproc-metastore/compare/v0.1.0...v0.2.0) (2021-06-02)


### Features

* add v1 ([#28](https://www.github.com/googleapis/python-dataproc-metastore/issues/28)) ([4d054d9](https://www.github.com/googleapis/python-dataproc-metastore/commit/4d054d92fed4296883e5ae09b99d57bd74d68fb4))

## 0.1.0 (2021-03-15)


### Features

* generate v1alpha ([2c025f8](https://www.github.com/googleapis/python-dataproc-metastore/commit/2c025f80c7791ef864ce2bf655429e1ecf40d288))
* generate v1beta ([942ddcd](https://www.github.com/googleapis/python-dataproc-metastore/commit/942ddcd6ddd18bd6d79cf2c57685a743ea35a376))


owl-bot-staging/v1alpha/noxfile.py
# -*- coding: utf-8 -*-
# Copyright 2022 Google LLC
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#     http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.
#
import os
import pathlib
import shutil
import subprocess
import sys


import nox  # type: ignore

ALL_PYTHON = [
    "3.6",
    "3.7",
    "3.8",
    "3.9",
    "3.10",
]

CURRENT_DIRECTORY = pathlib.Path(__file__).parent.absolute()

LOWER_BOUND_CONSTRAINTS_FILE = CURRENT_DIRECTORY / "constraints.txt"
PACKAGE_NAME = subprocess.check_output([sys.executable, "setup.py", "--name"], encoding="utf-8")

BLACK_VERSION = "black==19.10b0"
BLACK_PATHS = ["docs", "google", "tests", "samples", "noxfile.py", "setup.py"]
DEFAULT_PYTHON_VERSION = "3.9"

nox.sessions = [
    "unit",
    "cover",
    "mypy",
    "check_lower_bounds"
    # exclude update_lower_bounds from default
    "docs",
    "blacken",
    "lint",
    "lint_setup_py",
]

@nox.session(python=ALL_PYTHON)
def unit(session):
    """Run the unit test suite."""

    session.install('coverage', 'pytest', 'pytest-cov', 'asyncmock', 'pytest-asyncio')
    session.install('-e', '.')

    session.run(
        'py.test',
        '--quiet',
        '--cov=google/cloud/metastore_v1alpha/',
        '--cov=tests/',
        '--cov-config=.coveragerc',
        '--cov-report=term',
        '--cov-report=html',
        os.path.join('tests', 'unit', ''.join(session.posargs))
    )


@nox.session(python=DEFAULT_PYTHON_VERSION)
def cover(session):
    """Run the final coverage report.
    This outputs the coverage report aggregating coverage from the unit
    test runs (not system test runs), and then erases coverage data.
    """
    session.install("coverage", "pytest-cov")
    session.run("coverage", "report", "--show-missing", "--fail-under=100")

    session.run("coverage", "erase")


@nox.session(python=ALL_PYTHON)
def mypy(session):
    """Run the type checker."""
    session.install('mypy', 'types-pkg_resources')
    session.install('.')
    session.run(
        'mypy',
        '--explicit-package-bases',
        'google',
    )


@nox.session
def update_lower_bounds(session):
    """Update lower bounds in constraints.txt to match setup.py"""
    session.install('google-cloud-testutils')
    session.install('.')

    session.run(
        'lower-bound-checker',
        'update',
        '--package-name',
        PACKAGE_NAME,
        '--constraints-file',
        str(LOWER_BOUND_CONSTRAINTS_FILE),
    )


@nox.session
def check_lower_bounds(session):
    """Check lower bounds in setup.py are reflected in constraints file"""
    session.install('google-cloud-testutils')
    session.install('.')

    session.run(
        'lower-bound-checker',
        'check',
        '--package-name',
        PACKAGE_NAME,
        '--constraints-file',
        str(LOWER_BOUND_CONSTRAINTS_FILE),
    )

@nox.session(python=DEFAULT_PYTHON_VERSION)
def docs(session):
    """Build the docs for this library."""

    session.install("-e", ".")
    session.install("sphinx==4.0.1", "alabaster", "recommonmark")

    shutil.rmtree(os.path.join("docs", "_build"), ignore_errors=True)
    session.run(
        "sphinx-build",
        "-W",  # warnings as errors
        "-T",  # show full traceback on exception
        "-N",  # no colors
        "-b",
        "html",
        "-d",
        os.path.join("docs", "_build", "doctrees", ""),
        os.path.join("docs", ""),
        os.path.join("docs", "_build", "html", ""),
    )


@nox.session(python=DEFAULT_PYTHON_VERSION)
def lint(session):
    """Run linters.

    Returns a failure if the linters find linting errors or sufficiently
    serious code quality issues.
    """
    session.install("flake8", BLACK_VERSION)
    session.run(
        "black",
        "--check",
        *BLACK_PATHS,
    )
    session.run("flake8", "google", "tests", "samples")


@nox.session(python=DEFAULT_PYTHON_VERSION)
def blacken(session):
    """Run black. Format code to uniform standard."""
    session.install(BLACK_VERSION)
    session.run(
        "black",
        *BLACK_PATHS,
    )


@nox.session(python=DEFAULT_PYTHON_VERSION)
def lint_setup_py(session):
    """Verify that setup.py is valid (including RST check)."""
    session.install("docutils", "pygments")
    session.run("python", "setup.py", "check", "--restructuredtext", "--strict")
